compiler/damlc/tests/daml-test-files/ModuleImports_Main.daml

```diff
@@ -0,0 +1,11 @@
+-- Copyright (c) 2021, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+-- This test checks that we produce an $$imports value including all modules from which this one imports instances.
+
+-- @ERROR range=14:10-14:23; • No explicit implementation for; either ‘-’ or ‘negate’; • In the instance declaration for ‘Additive Expr’
+-- @QUERY-LF .modules[]
```

Comment:
I'm not sure what's the best way to produce the query I want here. Basically I want to check that `ModuleImports_OrphanInstance` is in the `$$imports` value.

Reply:
The queries are produced by JQ. To get the JSON you can change https://github.com/digital-asset/daml/blob/6bf45a344a07568f1c2564ba894860dd9b276129/compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs#L171 to an absolute directory (otherwise it ends up in the bazel sandbox and is annoying to find). That should allow you to run a reasonable query on it it. As an example, maybe look at the one for fundeps https://github.com/digital-asset/daml/blob/6bf45a344a07568f1c2564ba894860dd9b276129/compiler/damlc/tests/daml-test-files/FunctionalDependencies.daml#L8.

If you can’t find the right query, I’m happy to help.

Reply:
Thanks! I'll start with this :)